generators/pkg/generator/generate.go

```diff
@@ -194,6 +209,148 @@ func GeneratePipeline(github *GitHub) (*v1beta1.Pipeline, error) {
 	return pipeline, nil
 }
 
+// Generate the pipelinerun with the given generated pipeline
+func GeneratePipelineRun(p *v1beta1.Pipeline, g *GitHub) *v1beta1.PipelineRun {
+	if p.Namespace == "" {
+		p.Namespace = "default"
+	}
+
+	var value int64
+	var format resource.Format
+	if g.Spec.Storage != "" {
+		diskSize := resource.MustParse(g.Spec.Storage)
+		value = diskSize.Value()
+		format = diskSize.Format
+	} else {
+		value = 1024 * 1024 * 1024
+		format = resource.BinarySI
+	}
+
+	if g.Spec.Revision == "" {
+		g.Spec.Revision = "master"
+	}
+
+	return &v1beta1.PipelineRun{
+		TypeMeta: metav1.TypeMeta{
+			APIVersion: v1beta1.SchemeGroupVersion.String(),
+			Kind:       "PipelineRun",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			Namespace: p.Namespace,
+			Name:      p.Name + "-run",
+			Labels:    p.Labels,
+		},
+		Spec: v1beta1.PipelineRunSpec{
+			PipelineRef: &v1beta1.PipelineRef{
+				Name: p.Name,
+			},
+			Params: []v1beta1.Param{
+				{
+					Name: "gitrepositoryurl",
+					Value: v1beta1.ArrayOrString{
+						Type:      v1beta1.ParamTypeString,
+						StringVal: g.Spec.URL,
+					},
+				},
+				{
+					Name: "gitrevision",
+					Value: v1beta1.ArrayOrString{
+						Type:      v1beta1.ParamTypeString,
+						StringVal: g.Spec.Revision,
+					},
+				},
+			},
+			Workspaces: []v1beta1.WorkspaceBinding{
+				{
+					Name: p.Spec.Workspaces[0].Name,
```

Comment:
Are we guaranteed that this will always be present? We may want to add validation to check that this and other fields exist.

Reply:
To provide more context, if workspaces aren't set, this is going to cause the program to panic and crash.

Reply:
I think this workspace is present because I wire up workspaces for the GitHub repo in the generated task and pipeline. For other fields to check, maybe we can add the validation at parsing? For example, when parsing the input config, if users don't provide the url which is a required field, we could print an error.